Update WordPress doc for Bedrock and add some more informations about it (#838)
Scalingo/documentation: _posts/languages/php/2000-01-01-wordpress.md

---
title: Deploying WordPress on Scalingo
nav: WordPress
modified_at: 2020-01-29 00:00:00
tags: php, http, framework, wordpress, deployment
index: 99
---

## Detection

> When a PHP application is deployed, we're looking if the `wp-settings.php` file

Reviewer: ```suggestion
When a PHP application is deployed, Scalingo checks the existence of the `wp-settings.php` file at the root folder of your app.
```

is present at the root folder of your app.

> During the deployment process you'll see the following output, mentioning that

Reviewer: ```suggestion
During the deployment process, you'll see the following output mentioning that
```

> the framework has correctly been detected.

Reviewer: ```suggestion
the framework has correctly been detected:
```

> ```bash

Reviewer: ```suggestion
```text
```

-----> Detected WordPress
-----> Setting up WordPress
...
```

Or for Bedrock WordPress:

> ```bash

Reviewer: ```suggestion
```text
```

-----> Detected Bedrock WordPress
-----> Setting up Bedrock WordPress
...
```

## Bedrock: a Scalingo Friendly WordPress Boilerplate

WordPress is not well suited to be directly deployed on Scalingo. They do not follow the best modern practices
of web development such as [12 factor](https://12factor.net/).

The easiest way to get started with WordPress on Scalingo is to click on this button:

[![Deploy on
Scalingo](https://cdn.scalingo.com/deploy/button.svg)](https://my.osc-fr1.scalingo.com/deploy?source=https://github.com/Scalingo/scalingo-wordpress)

This one-click deploy button uses this [Scalingo Distribution](https://github.com/Scalingo/scalingo-wordpress).
It is based on [Bedrock](https://roots.io/bedrock/), and install everything for your WordPress
to work perfectly on a modern platform like Scalingo.

### Customize the Scalingo Distribution

> You may need to customize a bit the above-mentioned distribution in order to add a plugin or a theme for instance.

Reviewer: ```suggestion
You may need to customize a bit the above-mentioned distribution in order to add a plugin or a theme for instance. Follow these instructions to get started:
```

> 1. Clone the distribution

Reviewer: ```suggestion
1. Clone the distribution:
```

>    In order to do that, you first need to clone the distribution:

Reviewer: ```suggestion
```

   ```bash
   git clone https://github.com/Scalingo/scalingo-wordpress
   cd scalingo-wordpress
   ```

2. Create the application on Scalingo

>    Create the application though the Dashboard with a MySQL addon or with the [Scalingo CLI](http://cli.scalingo.com):

Reviewer: ```suggestion
   Create the application through the dashboard with a MySQL addon or with the [Scalingo CLI](http://cli.scalingo.com):
```

   ```bash

>    scalingo create my-wordpress

Reviewer: ```suggestion
   scalingo create my-app
```

all the doc contains `my-app` when using an app name

   scalingo addons-add mysql mysql-sandbox
   ```

> 3. Create an S3 Bucket on AWS and configure IAM user correctly

Reviewer: ```suggestion
3. Create a S3 Bucket on AWS and configure IAM user correctly
```

   IAM user security policy example, with required actions:

   ```bash
    {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Action": [
            "s3:PutObject",
            "s3:PutObjectAcl",
            "s3:PutObjectAclVersion",
            "s3:AbortMultipartUpload",
            "s3:ListBucket",
            "s3:GetObject"
          ],
          "Effect": "Allow",
          "Resource": "arn:aws:s3:::BUCKETNAME-HERE"
        },
        {
          "Action": [
            "s3:PutObject",
            "s3:PutObjectAcl",
            "s3:PutObjectAclVersion",
            "s3:AbortMultipartUpload",
            "s3:ListBucket",
            "s3:GetObject"
          ],
          "Effect": "Allow",
          "Resource": "arn:aws:s3:::BUCKETNAME-HERE/*"
        }
      ]
    }
   ```

4. Update application environment variables

>    Then, update your application environment through the Dashboard or with the

Reviewer: ```suggestion
   Then, update your application environment through the dashboard or with the
```

   [Scalingo CLI](http://cli.scalingo.com) `scalingo env-set VARIABLE_NAME=VALUE`:

   * `DATABASE_URL`: Connection string to the MySQL database - `mysql://localhost:3306/wp-bedrock` - Automatically added with the Scalingo MySQL addon
   * `WP_ENV`: Set to environment (`development`, `staging`, `production`)
   * `WP_HOME`: Full URL to WordPress home (https://my-wordpress.osc-fr1.scalingo.io)
   * `WP_SITEURL`: Full URL to WordPress including subdirectory (https://my-wordpress.osc-fr1.scalingo.io/wp)
   * `S3_UPLOADS_BUCKET`: Name of the S3 bucket to upload files to
   * `S3_UPLOADS_KEY`: AWS Access Key ID for S3 authentication
   * `S3_UPLOADS_SECRET`: AWS Secret Key for S3 authentication
   * `S3_UPLOADS_REGION`: Region of the S3 bucket
   * `AUTH_KEY`, `SECURE_AUTH_KEY`, `LOGGED_IN_KEY`, `NONCE_KEY`, `AUTH_SALT`, `SECURE_AUTH_SALT`, `LOGGED_IN_SALT`, `NONCE_SALT`

   You can get some random salts on the [Roots WordPress Salt Generator](https://roots.io/salts.html).

5. Add themes in `web/app/themes` as you would for a normal WordPress site.

6. Add plugins using [Composer](https://getcomposer.org/) and [WordPress Packagist](https://wpackagist.org/search?q=&type=plugin&search=)

>    Example for add the `Akismet` plugin:

Reviewer: ```suggestion
   Example to add the `Akismet` plugin:
```

   ```bash
   composer require --ignore-platform-reqs wpackagist-plugin/akismet
   ```

7. Deploy the application on Scalingo

   ```bash

>    # Optionally add theme to your git repository

Reviewer: I would add these instructions in the 5th point and not here

   git add web/app/themes
   git commit -m "Add themes"

>    # Then push to Scalingo

Reviewer: Comment is useless I think
```suggestion
```

   git push scalingo master
   ```

> 8. Access WP admin at `https://my-wordpress.osc-fr1.scalingo.io/wp/wp-admin`

Reviewer: ```suggestion
8. Access WP admin at `https://my-app.osc-fr1.scalingo.io/wp/wp-admin`
```

9. Activate the `S3 Uploads` plugin on WP Admin plugins page.

> 10. That's it.

Reviewer: ```suggestion
```

I don't really like this 10th point as there are already so many points to successfully deploy the WordPress
